insights/tests/client/test_client.py

```diff
@@ -18,3 +26,155 @@ def test_version():
+        InsightsConfig(unregister=True, analyze_container=True)
+
+
+def test_force_reregister_container():
+    with pytest.raises(ValueError):
+        InsightsConfig(reregister=True, analyze_container=True)
+
+
+def test_reg_check_registered():
+    # register the machine first
+    config = InsightsConfig(register=True)
+    client = InsightsClient(config)
+    try_auto_configuration(config)
+    assert client.register() is True
+    config.register = False
+
+    # test function and integration in .register()
+    assert client.get_registration_information()['status'] is True
+    assert client.register() is True
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is True
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is False
+
+
+def test_reg_check_unregistered():
+    # unregister the machine first
+    config = InsightsConfig(unregister=True)
+    client = InsightsClient(config)
+    try_auto_configuration(config)
+    assert client.unregister() is True
+    config.unregister = False
+
+    # test function and integration in .register()
+    assert client.get_registration_information()['status'] is False
+    assert client.register() is False
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is False
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is True
+
+
+def test_reg_check_registered_unreachable():
+    # register the machine first
+    config = InsightsConfig(register=True)
+    client = InsightsClient(config)
+    try_auto_configuration(config)
+    assert client.register() is True
+    config.register = False
+    config.http_timeout = 1
+
+    # set net delay and try to check registration
+    subprocess.call(
+        shlex.split('tc qdisc add dev eth0 root netem delay 1001ms'))
+    with pytest.raises(requests.exceptions.Timeout):
+        assert client.get_registration_information()['unreachable'] is True
+        assert client.register() is None
+    subprocess.call(
+        shlex.split('tc qdisc del dev eth0 root netem delay 1001ms'))
+    for r in constants.registered_files:
+        assert os.path.isfile(r) is True
+    for u in constants.unregistered_files:
+        assert os.path.isfile(u) is False
+
+
+def test_reg_check_unregistered_unreachable():
+    # unregister the machine first
+    config = InsightsConfig(unregister=True)
+    client = InsightsClient(config)
+    try_auto_configuration(config)
+    assert client.unregister() is True
+    config.unregister = False
+    config.http_timeout = 1
+
+    # set net delay and try to check registration
+    subprocess.call(
+        shlex.split('tc qdisc add dev eth0 root netem delay 1001ms'))
+    with pytest.raises(requests.exceptions.Timeout):
+        assert client.get_registration_information()['unreachable'] is True
+        assert client.register() is None
+    subprocess.call(
+        shlex.split('tc qdisc del dev eth0 root netem delay 1001ms'))
```

Comment:
I don't think this fits in a unit test.  Pretty sure this requires root to run correctly.

Reply:
Hmm alright. Wonder if there's a better way to force a timeout